src/container.cpp

```diff
@@ -42,34 +41,33 @@ Container::~Container()
 	if (getID() == ITEM_BROWSEFIELD) {
 		g_game.browseFields.erase(getTile());
 
-		for (Item* item : itemlist) {
+		for (const auto& item : itemList) {
 			item->setParent(parent);
 		}
 	} else {
-		for (Item* item : itemlist) {
+		for (const auto& item : itemList) {
 			item->setParent(nullptr);
-			item->decrementReferenceCounter();
 		}
 	}
 }
 
-Item* Container::clone() const
+std::shared_ptr<Item> Container::clone() const
 {
-	Container* clone = static_cast<Container*>(Item::clone());
+	auto clone = std::static_pointer_cast<Container>(Item::clone());
```

Comment:
If I recall correctly, it has the same effect and centralizes the use of `static_cast` in `getX`.
```suggestion
	auto clone = Item::clone()->getContainer();
```

Reply:
On second thought, maybe we get a perform gain here?

Reply:
Could this be `const`
```suggestion
	cont auto clone = std::static_pointer_cast<Container>(Item::clone());
```

Reply:
`Item::clone()` can return `nullptr`, which can be `static_cast`ed but not `->getContainer()`d

Both result in the same code